wagtail/blocks/field_block.py

```diff
@@ -831,7 +832,7 @@ def clean(self, value):
         return super().clean(value)
 
     def extract_references(self, value):
-        if value is not None:
+        if value is not None and issubclass(self.model_class, Model):
```

Comment:
Not sure if it has the same problem as above, but if so then we'd have to check `model_class` too

```suggestion
        if value is not None and self.model_class and issubclass(self.model_class, Model):
```

Reply:
I don't think this is necessary - `model_class` is used so widely within ChooserBlock to do the various type conversions that you'd have to basically override the entire implementation to make it work without a model class (and at that point, one extra method to override isn't going to make a difference).